src/ucp/core/ucp_ep.c

```diff
@@ -309,6 +309,45 @@ ucp_ep_adjust_params(ucp_ep_h ep, const ucp_ep_params_t *params)
     return UCS_OK;
 }
 
+ucs_status_t ucp_ep_perf_query(ucp_ep_h ep, ucp_ep_perf_attr_t *attr)
+{
+    const ucp_worker_h worker      = ep->worker;
+    const ucp_context_h context    = worker->context;
+    const ucp_ep_config_key_t *key = &ucp_ep_config(ep)->key;
+    ucp_lane_index_t lane;
+    ucp_worker_iface_t *wiface;
+    uct_iface_attr_t *iface_attr;
+    ucp_rsc_index_t rsc_index = 0, max_bandwidth_rsc_index = 0;
+    double bandwidth = 0, max_bandwidth = 0;
+
+    if ((attr->field_mask & UCP_EP_PERF_ATTR_FIELD_ESTIMATED_TIME) &&
+        (attr->field_mask & UCP_EP_PERF_ATTR_FIELD_MESSAGE_SIZE)) {
+        for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
+            if (lane == key->cm_lane) {
+                /* Skip CM lanes for banwidth calculation */
+                continue;
+            }
+
+            rsc_index       = key->lanes[lane].rsc_index;
+            wiface          = worker->ifaces[rsc_index];
+            bandwidth       = ucp_tl_iface_bandwidth(context,
+                                                     &wiface->attr.bandwidth);
```

Comment:
weird indent

Reply:
Why? The indent is on the first char after the `(`.

Reply:
i mean, too many spaces before =